bin/peggy-cli.js

```diff
@@ -629,26 +639,40 @@ class PeggyCLI extends Command {
    * @returns {Promise<number>}
    */
   async run() {
-    let inputStream = undefined;
-
-    if (this.inputFile === "-") {
-      this.std.in.resume();
-      inputStream = this.std.in;
-      this.argv.grammarSource = "stdin";
-    } else {
-      this.argv.grammarSource = this.inputFile;
-      inputStream = fs.createReadStream(this.inputFile);
-    }
+    const sources = [];
 
     let exitCode = 1;
     let errorText = "";
-    let input = "";
+    let prevSource = process.cwd() + "/";
     try {
-      this.verbose("CLI", errorText = "reading input stream");
-      input = await readStream(inputStream);
+      for (const source of this.inputFiles) {
+        const input = { source, text: null };
+        this.verbose("CLI", errorText = `reading input "${source}"`);
+        if (source === "-") {
+          input.source = "stdin";
+          this.std.in.resume();
+          input.text = await readStream(this.std.in);
+        } else if (source.startsWith("npm:")) {
+          const req = (
+            // In node 12+, createRequire is documented.
+            // In node 10, createRequireFromPath is the least-undocumented approach.
+            Module.createRequire || Module.createRequireFromPath
```

Comment:
Using nullish coalescing `??` would be more idiomatic here. Are we on a node version that doesn't support it? Not seeing it anywhere in the source.

Reply:
I think createRequire is in node12, but I'm fine bumping to Node18 required, and using anything new we want in the CLI code.  The web-facing code we should be a little more careful with.

Reply:
Just removed node 10 compatibility.